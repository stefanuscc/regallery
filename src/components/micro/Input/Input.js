import { useCallback } from 'react'
import styles from './Input.module.scss'

export const Input = (props) => {
    const {
        className="",
        value, 
        onChange,
        ...attrs
    } = props

    const classNames = `${styles.field} ${className}`
    const handleChange = useCallback((event) => {
        onChange(event.target.value)
    }, [onChange])

    return (
        <input {...attrs} className={classNames} value={value} onChange={handleChange}/>
    )
}

export default Input