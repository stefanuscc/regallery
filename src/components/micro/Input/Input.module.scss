// Vars
@import "/src/_vars";

.field {
    outline: none;
    border: 1px solid #ccc;
    padding: .6rem .8rem;
    font-family: 'Poppins', sans-serif;
}