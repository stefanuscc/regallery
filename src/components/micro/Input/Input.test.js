import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Input from './Input';
import "../../../setupTests"

test('Input to match snapshot', () => {
    const component = renderer.create(
        <Input type="text" className="" placeholder="Rename Image" value="Image #1" onChange={(val) => alert(val) }/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Input value is persistence', () => {
    const input = shallow(<Input type="text" className="" placeholder="Rename Image" value="Image #1" onChange={() => {}}/>);

    expect(input.prop('value')).toEqual('Image #1');

    // input.simulate('change', {
    //     target: {
    //         value: 'Mountain View'
    //     }
    // });

    // expect(input.prop('value')).toEqual('Mountain View');
});