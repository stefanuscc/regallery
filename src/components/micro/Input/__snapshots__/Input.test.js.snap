// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input to match snapshot 1`] = `
<input
  className="field "
  onChange={[Function]}
  placeholder="Rename Image"
  type="text"
  value="Image #1"
/>
`;
